packages/move/sources/vesting.move

```diff
@@ -25,14 +25,29 @@ module deployment_addr::vesting {
     /// User try to claim zero
     const ERR_AMOUNT_ZERO: u64 = 10;
 
-    struct VestingStream has key, store, drop {
+    // FIX: key ability is not needed for VestingStream since it isn't keyed to an address in global storage
+
+    // FIX: current implementation treats cliff as a delay period before linear vesting begins. Which is kind of redundant since the start_time can be specified
+    // For a proper vesting cliff implementation, VestingStream should include a cliff_amount field
+    // that represents tokens immediately unlocked when cliff is reached. Current implementation
+    // seems to always start linear vesting from 0 at cliff time, whereas typically some percentage
+    // of tokens should unlock immediately at cliff time(in your case that percentage is effectively 0).
```

Comment:
This makes sense, the cliff and start_time can be combined to a single field (start_time), and a new cliff_amount should be added. The calculation of vested_amount should be adjusted to account for the cliff_amount.

Reply:
yes